monai/apps/detection/transforms/dictionary.py

```diff
@@ -1174,9 +1174,8 @@ def randomize(  # type: ignore
 
     def __call__(self, data: Mapping[Hashable, torch.Tensor]) -> List[Dict[Hashable, torch.Tensor]]:
         d = dict(data)
-        spatial_dims = len(d[self.image_keys[0]].shape) - 1
         image_size = d[self.image_keys[0]].shape[1:]
-        self.spatial_size = ensure_tuple_rep(self.spatial_size_, spatial_dims)
+        self.spatial_size = fall_back_tuple(self.spatial_size_, image_size)
```

Comment:
This fix refers to: https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/croppad/array.py#L1094

Thanks.